src/anyp/Uri.cc

```diff
@@ -857,64 +856,73 @@ urlCheckRequest(const HttpRequest * r)
      */
 
     if (r->method == Http::METHOD_CONNECT)
-        return 1;
+        return true;
 
     // we support OPTIONS and TRACE directed at us (with a 501 reply, for now)
     // we also support forwarding OPTIONS and TRACE, except for the *-URI ones
     if (r->method == Http::METHOD_OPTIONS || r->method == Http::METHOD_TRACE)
         return (r->header.getInt64(Http::HdrType::MAX_FORWARDS) == 0 || r->url.path() != AnyP::Uri::Asterisk());
 
     if (r->method == Http::METHOD_PURGE)
-        return 1;
+        return true;
 
     /* does method match the protocol? */
     switch (r->url.getScheme()) {
 
     case AnyP::PROTO_URN:
+        return true;
 
     case AnyP::PROTO_HTTP:
+        return true;
 
     case AnyP::PROTO_CACHE_OBJECT:
-        rc = 1;
-        break;
-
-    case AnyP::PROTO_FTP:
+        return true;
 
-        if (r->method == Http::METHOD_PUT)
-            rc = 1;
+    case AnyP::PROTO_FTP:        
+        if (r->method == Http::METHOD_PUT ||
+            r->method == Http::METHOD_GET ||
+            r->method == Http::METHOD_HEAD )
+            return true;
+        return false;
```

Comment:
```suggestion
        return r->method == Http::METHOD_PUT || r->method == Http::METHOD_GET || r->method == Http::METHOD_HEAD;
```

I do not insist on this change.

Reply:
I'll skip this one, I think it's more readable on multiple lines